Toolkit/ArcGISToolkit/TimeSlider.swift

```diff
@@ -0,0 +1,2603 @@
+        layer.addSublayer(lowerThumbLayer)
+        
+        // Add upper thumb layer
+        upperThumbLayer.timeSlider = self
+        upperThumbLayer.contentsScale = UIScreen.main.scale
+        layer.addSublayer(upperThumbLayer)
+        
+        // Add the minimum value label
+        fullExtentStartTimeLabel.isHidden = !(fullExtentLabelsVisible && isSliderVisible)
+        fullExtentStartTimeLabel.foregroundColor = fullExtentLabelColor.cgColor
+        fullExtentStartTimeLabel.alignmentMode = kCAAlignmentCenter
+        fullExtentStartTimeLabel.frame = CGRect.zero
+        fullExtentStartTimeLabel.contentsScale = UIScreen.main.scale
+        fullExtentStartTimeLabel.fontSize = fullExtentLabelFont.pointSize
+        layer.addSublayer(fullExtentStartTimeLabel)
+        
+        // Add the minimum value label
+        fullExtentEndTimeLabel.isHidden = !(fullExtentLabelsVisible && isSliderVisible)
+        fullExtentEndTimeLabel.foregroundColor = fullExtentLabelColor.cgColor
+        fullExtentEndTimeLabel.alignmentMode = kCAAlignmentCenter
+        fullExtentEndTimeLabel.frame = CGRect.zero
+        fullExtentEndTimeLabel.contentsScale = UIScreen.main.scale
+        fullExtentEndTimeLabel.fontSize = fullExtentLabelFont.pointSize
+        layer.addSublayer(fullExtentEndTimeLabel)
+        
+        // Add the lower value label
+        currentExtentStartTimeLabel.isHidden = !(labelMode == .thumbs && isSliderVisible)
+        currentExtentStartTimeLabel.foregroundColor = currentExtentLabelColor.cgColor
+        currentExtentStartTimeLabel.alignmentMode = kCAAlignmentCenter
+        currentExtentStartTimeLabel.frame = CGRect.zero
+        currentExtentStartTimeLabel.contentsScale = UIScreen.main.scale
+        currentExtentStartTimeLabel.fontSize = currentExtentLabelFont.pointSize
+        layer.addSublayer(currentExtentStartTimeLabel)
+        
+        // Add the upper value label
+        currentExtentEndTimeLabel.isHidden = !(labelMode == .thumbs && isSliderVisible)
+        currentExtentEndTimeLabel.foregroundColor = currentExtentLabelColor.cgColor
+        currentExtentEndTimeLabel.alignmentMode = kCAAlignmentCenter
+        currentExtentEndTimeLabel.frame = CGRect.zero
+        currentExtentEndTimeLabel.contentsScale = UIScreen.main.scale
+        currentExtentEndTimeLabel.fontSize = currentExtentLabelFont.pointSize
+        layer.addSublayer(currentExtentEndTimeLabel)
+        
+        // Create the images
+        let bundle = Bundle(for: type(of: self))
+        playImage = UIImage(named: "Play", in: bundle, compatibleWith: nil)
+        pauseImage = UIImage(named: "Pause", in: bundle, compatibleWith: nil)
+        forwardImage = UIImage(named: "Forward", in: bundle, compatibleWith: nil)
+        backImage = UIImage(named: "Back", in: bundle, compatibleWith: nil)
+        
+        // Add Play/Pause button
+        let p = UIButton(type: .custom)
+        p.setBackgroundImage(playImage, for: .normal)
```

Comment:
@njarecha You could save yourself some work by taking advantage of the fact that each state can have a separate image. For example, say that `normal` is the paused state (play image will be shown) and that `selected` is the playing state (pause image will be shown). You would only need to set the image for the two different states here and then update which image is shown by setting the button's state elsewhere.

Reply:
fixed